sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneExceptionExtensionsTests.cs

```diff
@@ -0,0 +1,102 @@
+    ///   class.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Parallelizable(ParallelScope.Children)]
+    public class TrackOneExceptionExtensionsTests
+    {
+        /// <summary>
+        ///   The set of test cases for the well-known types derrived from the <see cref="EventHubsException" />
+        ///   and their expected transient status.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ExceptionMappingTestCases()
+        {
+            TrackOne.EventHubsException exception;
+
+            exception = new TrackOne.EventHubsCommunicationException("One");
+            exception.EventHubsNamespace = "test_thing";
+            yield return new object[] {exception, typeof(Errors.EventHubsCommunicationException) };
+
+            exception = new TrackOne.EventHubsTimeoutException("Two");
+            exception.EventHubsNamespace = "OMG!-Thing!";
+            yield return new object[] { exception, typeof(Errors.EventHubsTimeoutException) };
+
+            yield return new object[] { new TrackOne.MessagingEntityNotFoundException("Three"), typeof(Errors.EventHubsResourceNotFoundException) };
+            yield return new object[] { new TrackOne.MessageSizeExceededException("Four"), typeof(Errors.MessageSizeExceededException) };
+            yield return new object[] { new TrackOne.QuotaExceededException("Five"), typeof(Errors.QuotaExceededException) };
+            yield return new object[] { new TrackOne.ReceiverDisconnectedException("Six"), typeof(Errors.ConsumerDisconnectedException) };
+            yield return new object[] { new TrackOne.ServerBusyException("Seven"), typeof(Errors.ServiceBusyException) };
+            yield return new object[] { new TrackOne.EventHubsException(true, "Eight"), typeof(Errors.EventHubsException) };
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="TrackOneExceptionExtensions.MapException" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void MapExceptionValidatesTheInstance()
+        {
+            Assert.That(() => ((TrackOne.EventHubsException)null).MapException(), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies that well-known derrived exception types have the correct value for their
+        ///   <see cref="EventHubsException.IsTransient" /> property.
+        /// </summary>
+        ///
+        [Test]
+        [TestCaseSource(nameof(ExceptionMappingTestCases))]
+        public void DerrivedExceptionsHaveTheCorrectTransientValues(Exception exception,
+                                                                    Type expectedMappedType)
+        {
+            var eventHubsException = (TrackOne.EventHubsException)exception;
+            var mappedException = eventHubsException.MapException();
+
+            Assert.That(mappedException, Is.Not.Null, "The mapping should produce an exception.");
```

Comment:
Is there a reason/advantage to use `Assert.That` instead of directly calling `Assert.NotNull()`

Reply:
In my opinion, it helps improve readability because it reads more like a sentence.  It's not quite a true BDD-style assert (such as FluentAssertions or Chai would provide) , but it doesn't require another dependency nor create as much friction for those that prefer to use the old-school `Assert.XXX` pattern.